[CI] GenerativeIT test failing

**Build Scans:**
- [elasticsearch-periodic #9845 / release-tests](https://gradle-enterprise.elastic.co/s/377qopstdlwqg)
- [elasticsearch-periodic #9817 / release-tests](https://gradle-enterprise.elastic.co/s/havsl2r7ja75m)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT.test" -Dtests.seed=C7D3725E3A2262BA -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=frr -Dtests.timezone=Australia/Hobart -Druntime.java=24
```

**Applicable branches:**
9.1

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from c*,airports_no_doc_*,n* | limit 1615| rename scalerank as language_code | lookup join languages_lookup on language_code | where false | rename `abbrev` AS CFniwsvPEXy, `rgb_vector` AS RAUILybcyw
exception: method [POST], host [http://[::1]:36063], URI [/_query?format=smile&pretty=true&error_trace=true], status line [HTTP/1.1 400 Bad Request]
:)
úerrorúroot_causeøútypeUverification_exceptionreasonàFound 1 problem
line 1:174: Cannot use field [rgb_vector] with unsupported type [dense_vector]üstack_traceäorg.elasticsearch.xpack.esql.VerificationException: Found 1 problem
line 1:174: Cannot use field [rgb_vector] with unsupported type [dense_vector]
	at org.elasticsearch.xpack.esql.analysis.Analyzer.verify(Analyzer.java:205)
	at org.elasticsearch.xpack.esql.analysis.Analyzer.analyze(Analyzer.java:199)
	at org.elasticsearch.xpack.esql.session.EsqlSession.lambda$analyzedPlan$9(EsqlSession.java:405)
	at org.elasticsearch.xpack.esql.session.EsqlSessi
[truncated]
```

**Issue Reasons:**
- [9.1] 2 failures in test test (1.8% fail rate in 113 executions)
- [9.1] 2 failures in step release-tests (40.0% fail rate in 5 executions)
- [9.1] 2 failures in pipeline elasticsearch-periodic (40.0% fail rate in 5 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java
@@ -38,7 +38,11 @@ import static org.elasticsearch.test.ESTestCase.randomLongBetween;
 
 public class EsqlQueryGenerator {
 
-    public record Column(String name, String type) {}
+    public static final String COLUMN_NAME = "name";
+    public static final String COLUMN_TYPE = "type";
+    public static final String COLUMN_ORIGINAL_TYPES = "original_types";
+
+    public record Column(String name, String type, List<String> originalTypes) {}
 
     public record QueryExecuted(String query, int depth, List<Column> outputSchema, List<List<Object>> result, Exception exception) {}
 
@@ -331,7 +335,7 @@ public class EsqlQueryGenerator {
             || field.name().equals("<no-fields>")
             // no dense vectors for now, they are not supported in most commands
             || field.type().contains("vector")
-            || field.type().contains("unsupported")) == false;
+            || field.originalTypes.stream().anyMatch(x -> x.contains("vector"))) == false;
     }
 
     public static String unquote(String colName) {

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java
@@ -330,7 +330,8 @@ public class EsqlQueryGenerator {
             // this is a known pathological case, no need to test it for now
             || field.name().equals("<no-fields>")
             // no dense vectors for now, they are not supported in most commands
-            || field.type().contains("vector")) == false;
+            || field.type().contains("vector")
+            || field.type().contains("unsupported")) == false;
     }
 
     public static String unquote(String colName) {

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
@@ -32,6 +32,9 @@ import static org.elasticsearch.xpack.esql.CsvTestsDataLoader.CSV_DATASET_MAP;
 import static org.elasticsearch.xpack.esql.CsvTestsDataLoader.ENRICH_POLICIES;
 import static org.elasticsearch.xpack.esql.CsvTestsDataLoader.availableDatasetsForEs;
 import static org.elasticsearch.xpack.esql.CsvTestsDataLoader.loadDataSetIntoEs;
+import static org.elasticsearch.xpack.esql.qa.rest.generative.EsqlQueryGenerator.COLUMN_NAME;
+import static org.elasticsearch.xpack.esql.qa.rest.generative.EsqlQueryGenerator.COLUMN_ORIGINAL_TYPES;
+import static org.elasticsearch.xpack.esql.qa.rest.generative.EsqlQueryGenerator.COLUMN_TYPE;
 
 public abstract class GenerativeRestTest extends ESRestTestCase {
 
@@ -212,11 +215,22 @@ public abstract class GenerativeRestTest extends ESRestTestCase {
 
     @SuppressWarnings("unchecked")
     private static List<EsqlQueryGenerator.Column> outputSchema(Map<String, Object> a) {
-        List<Map<String, String>> cols = (List<Map<String, String>>) a.get("columns");
+        List<Map<String, ?>> cols = (List<Map<String, ?>>) a.get("columns");
         if (cols == null) {
             return null;
         }
-        return cols.stream().map(x -> new EsqlQueryGenerator.Column(x.get("name"), x.get("type"))).collect(Collectors.toList());
+        return cols.stream()
+            .map(x -> new EsqlQueryGenerator.Column((String) x.get(COLUMN_NAME), (String) x.get(COLUMN_TYPE), originalTypes(x)))
+            .collect(Collectors.toList());
+    }
+
+    @SuppressWarnings("unchecked")
+    private static List<String> originalTypes(Map<String, ?> x) {
+        List<String> originalTypes = (List<String>) x.get(COLUMN_ORIGINAL_TYPES);
+        if (originalTypes == null) {
+            return List.of();
+        }
+        return originalTypes;
     }
 
     private List<String> availableIndices() throws IOException {

